### DISCUSSION TITLE
Make incrementQuietly and decrementQuietly public fuction

### DISCUSSION THREAD

**Discussion by Armanyes** [laravel/framework#43210] [General]
Today I tried to use decrementQuietly but as it is protected I couldn't. Is it possible to make it public function?

**Comment by henzeb** [comment]
Assuming this question is eloquent related ?

**Comment by Sophist-UK** [comment]
One obvious answer is to subclass the class these are in and do something like the following:
```
public function decrementQuietly(...$args)
{
    parent::decrementQuietly(...$args);
}
```

**Reply by henzeb** [reply]
But this is laravel we talk about. That's why I asked for a more descriptive question.

**Comment by eduance** [comment]
I agree as I personally think a developer should not want to implement this same logic twice which would violate the DRY pattern. Besides that I would say having to implement this logic in a project violates the single responsibility principle as a u user model for instance shouldn't have to be bothered with doing a database transaction quietly. I have made a PR to address this issue: https://github.com/laravel/framework/pull/43220

**Reply by Sophist-UK** [reply]
I agree that this would be better as a PR. but by all appearances PRs are difficult to get accepted for Laravel and they often are closed with boilerplate text that says that it should be done via a package instead.

**Reply by eduance** [reply]
Let's hope this is not one of them. :-) I do understand your point, as yes, Laravel denies a lot of PRs but they usually have a reasoning for it but it's worth trying atleast.

**Reply by Sophist-UK** [reply]
The argument against merging PRs is to stop the code base increasing substantially. This change to a public function does not do that, so I think you may have a better chance.

And whilst they do have a point, the creation of huge numbers of very small packages of variable quality to achieve minor enhancements to the main code base that could have been done via a PR brings its own issues of security, fragmentation and supportability.
